internal/manifests/collector/adapters/config_to_ports.go

```diff
@@ -52,119 +50,71 @@ func ConfigToExporterPorts(logger logr.Logger, config map[interface{}]interface{
 	// the exporter-qualifier is what comes after the slash in the exporter name, but typically nil
 	// examples:
 	// ```yaml
-	// exporters:
-	//   exampleexporter:
+	// components:
+	//   componentexample:
 	//     endpoint: 0.0.0.0:12345
-	//   exampleexporter/settings:
+	//   componentexample/settings:
 	//     endpoint: 0.0.0.0:12346
-	// in this case, we have two ports, named: "exampleexporter" and "exampleexporter-settings"
-	exportersProperty, ok := config["exporters"]
+	// in this case, we have 2 ports, named: "componentexample" and "componentexample-settings"
+	componentsProperty, ok := config[fmt.Sprintf("%ss", cType.String())]
 	if !ok {
-		return nil, ErrNoExporters
-	}
-	expEnabled := GetEnabledExporters(logger, config)
-	if expEnabled == nil {
-		return nil, ErrExportersNotAMap
+		return nil, fmt.Errorf("no %ss available as part of the configuration", cType)
 	}
-	exporters, ok := exportersProperty.(map[interface{}]interface{})
+
+	components, ok := componentsProperty.(map[interface{}]interface{})
 	if !ok {
-		return nil, ErrExportersNotAMap
+		return nil, fmt.Errorf("%ss doesn't contain valid components", cType.String())
+	}
+
+	var compEnabled map[interface{}]bool
+
+	switch cType {
+	case ComponentTypeExporter:
+		compEnabled = GetEnabledExporters(logger, config)
+	case ComponentTypeReceiver:
+		compEnabled = GetEnabledReceivers(logger, config)
+	}
+
+	if compEnabled == nil {
+		return nil, fmt.Errorf("no enabled %ss available as part of the configuration", cType)
 	}
 
 	ports := []corev1.ServicePort{}
-	for key, val := range exporters {
-		// This check will pass only the enabled exporters,
+	for key, val := range components {
+		// This check will pass only the enabled components,
 		// then only the related ports will be opened.
-		if !expEnabled[key] {
+		if !compEnabled[key] {
 			continue
 		}
 		exporter, ok := val.(map[interface{}]interface{})
 		if !ok {
-			logger.V(2).Info("exporter doesn't seem to be a map of properties", "exporter", key)
+			logger.V(2).Info("component doesn't seem to be a map of properties", cType.String(), key)
 			exporter = map[interface{}]interface{}{}
 		}
 
-		exprtName := key.(string)
-		exprtParser, err := exporterParser.For(logger, exprtName, exporter)
-		if err != nil {
-			logger.V(2).Info("no parser found for '%s'", exprtName)
-			continue
+		cmptName := key.(string)
+		var cmptParser parser.ComponentPortParser
+		var err error
+		switch cType {
+		case ComponentTypeExporter:
+			cmptParser, err = exporterParser.For(logger, cmptName, exporter)
+		case ComponentTypeReceiver:
+			cmptParser, err = receiverParser.For(logger, cmptName, exporter)
```

Comment:
Once config is no longer a string we should refactor this section as well IMO. I.e. once config has a defined `receivers` and `exporters` type it would be great if we didn't have to do this type of switching and instead put the methods on the structs instead.